crates/services/src/lib.rs

```diff
@@ -19,13 +19,27 @@ pub mod stream {
         core::pin::Pin<Box<dyn Stream<Item = T> + Send + Sync + 'static>>;
 }
 
+/// The source of some network data.
+pub struct SourcePeer<T> {
+    /// The source of the data.
+    pub peer_id: PeerId,
+    /// The data.
+    pub data: T,
+}
+
+/// Placeholder for a peer id
+pub struct PeerId;
```

Comment:
Yeh that is basically why I left this here as a placeholder because I wasn't sure what it should be. @leviathanbeak Do you know how long the bytes is for the PeerId? I could change this to `pub struct PeerId([u8; 32])` or whatever the length is.

Reply:
Looks like it's variable so will make this a vec